### PR TITLE
Work without EventMachine

### PR DESCRIPTION
**Pull request by gmalette**

### CODE REVIEW

File: spec/stream_action_spec.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+      EM.next_tick {
+        body.close
+        response.finish
+
+        expect(response.body).to eq(event.to_message)
+        EM.stop
+      }
+    end
+  end
```
Comment (gmalette) [inline spec/stream_action_spec.rb:74]: any idea how I cold do this better? using the Rack test methods, the body is closed as soon as headers are returned, not leaving enough time for the publish to go through.

**Comment by gmalette** [comment]
ping @byroot

**Comment by gmalette** [comment]
The PR still depends on EM being present (and is required by the gemspec), but the `EM.reactor_running?` could be prefixed by `defined?(EventMachine) && running?` Ithink

**Comment by gmalette** [comment]
To :tophat: you can `cd example` and `puma -p 3000` and `new EventSource('http://localhost:3000')`. Notice that the connection will be closed immediately, and subsequent connections will have an `Last-Event-ID` header, and will reconnect every 10s

**Comment by byroot** [comment]
:tophat:ed. Works like a charm!

**Pull request merged**